### PR TITLE
Create Seeding Script

### PR DESCRIPTION
**Pull request by daniel-sm-yu**
## Notion ticket link

[Database Seeding](https://www.notion.so/uwblueprintexecs/Task-Board-485f846ae00243599dddba2be9bcba82?p=18d5693d2f3145d390d4bca1b8bdc384)

## Implementation description
* created a script at `database/scripts/seed.ts` that populated collections with mock data
* moved model files into `database/models`
* renamed Module's `tool` to `toolID` and `slides` to `slideIDs`

## Steps to test
1. ensure `MONGODB_URL` and `MONGODB_DB` are set in `.env.local`
2. run `yarn seed --init` in the terminal (--init means it's the first time the script is being ran and won't try to drop collections)
3. run `yarn seed` for all subsequent seeding needs

## What should reviewers focus on?
* mock data matches schema

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: database/scripts/seed.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+            question: faker.lorem.words() + faker.lorem.words() + "?",
+            options: mockOptions[i % mockOptions.length],
+            questionNumber: (i % QUESTIONS_PER_GROUP) + 1,
+        });
+    }
+    return questions;
+}
+
+function mockGroups() {
+    const groups = [];
```
Comment (nimwijetunga) [inline database/scripts/seed.ts:85]: is there a reason we don't set the `_id` for a group here?

File: database/scripts/seed.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+    for (let i = 0; i < GROUP_COUNT; i++) {
+        const questionIDs = [];
+
+        for (let j = 0; j < QUESTIONS_PER_GROUP; j++) {
+            questionIDs.push(i * QUESTIONS_PER_GROUP + j);
```
Comment (nimwijetunga) [inline database/scripts/seed.ts:91]: what is this doing? Can you add a comment here please

File: database/scripts/seed.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+}
+
+function mockSlides() {
+    const slides = [];
+
+    for (let i = 0; i < SLIDE_COUNT; i++) {
+        const componentIDs = [];
+
+        for (let j = 0; j < COMPONENTS_PER_SLIDE; j++) {
+            componentIDs.push(i * COMPONENTS_PER_SLIDE + j);
```
Comment (nimwijetunga) [inline database/scripts/seed.ts:119]: add a comment on what this inner loop is doing

File: database/scripts/seed.ts [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+                    url: faker.internet.url(),
+                },
+                {
+                    title: faker.lorem.words(),
+                    description: faker.lorem.sentences(),
+                    url: faker.internet.url(),
+                },
+            ],
+            selfCheckGroupID: i < GROUP_COUNT ? i : null,
+            relatedToolsIDs: i > 0 ? [0] : null,
```
Comment (nimwijetunga) [inline database/scripts/seed.ts:185]: do we want to randomly pick some related tools just so we have some data for this?

**Review by nimwijetunga — APPROVED** [review]
lgtm! Just a few minor comments

**Review by mkanoria — APPROVED** [review]
LGTM!

File: database/scripts/seed.ts (hunk 1 above)
Comment (daniel-sm-yu) [reply database/scripts/seed.ts:85]: good catch! we'll need to set `_id` so we can link these groups with tools

File: database/scripts/seed.ts (hunk 4 above)
Comment (daniel-sm-yu) [reply database/scripts/seed.ts]: I randomized the `relatedToolsIDs` of each tool and left some comments

**Comment by mkanoria** [comment]
Thanks for updating this, you should be good to merge now. @nimwijetunga anything else to update?

**Pull request merged**